### PR TITLE
Move most swapchain extension checks from swapchain layer to PV

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

**Comment by chrisforbes** [comment]
Updated with require_*_extension helper functions, and a layer-wide rename of skip_call to skip.

**Comment by chrisforbes** [comment]
Rebased past some PV changes in master.

**Comment by chrisforbes** [comment]
Also moved the KHR_display enable checks across.

**Comment by chrisforbes** [comment]
@tobine if you're happy with the new changes, I think this is ready to go.

**Comment by chrisforbes** [comment]
This apparently still causes strange failures in windows debug. I don't have a suitable system to debug that, and inspection isn't getting me anywhere.

File: layers/parameter_validation.cpp [hunk 1]
```diff
@@ -4914,6 +4923,33 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateWin32SurfaceKHR(VkInstance instance, const
...
+
+    if (!instance_extension_map[get_dispatch_table(pc_instance_table_map, physicalDevice)].win32_enabled) {
+        skip_call |=
+            log_msg(my_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_PHYSICAL_DEVICE_EXT,
+                    reinterpret_cast<uint64_t>(physicalDevice), __LINE__, EXTENSION_NOT_ENABLED, LayerName,
+                    "vkGetPhysicalDeviceWin32PresentationSupportKHR() called even though the %s extension was not enabled for this VkInstance.",
+                    VK_KHR_WIN32_SURFACE_EXTENSION_NAME);
+    }
+
+    skip_call |= parameter_validation_vkGetPhysicalDeviceWin32PresentationSupportKHR(physicalDevice, queueFamilyIndex);
```
Comment (tobine) [inline layers/parameter_validation.cpp]: I believe the problem on Windows is that there is no definition for this function.

**Pull request merged**